src/main/java/io/confluent/connect/hdfs/HdfsSinkTask.java

```diff
@@ -49,14 +50,24 @@ public String version() {
   public void start(Map<String, String> props) {
     try {
       HdfsSinkConnectorConfig connectorConfig = new HdfsSinkConnectorConfig(props);
+      boolean hiveIntegration = connectorConfig.getBoolean(HdfsSinkConnectorConfig.HIVE_INTEGRATION_CONFIG);
+      if (hiveIntegration) {
+        Compatibility compatibility = SchemaUtils.getCompatibility(
+            connectorConfig.getString(HdfsSinkConnectorConfig.SCHEMA_COMPATIBILITY_CONFIG));
+        if (compatibility == Compatibility.NONE) {
```

Comment:
Should all these types of checks just be performed once in the connector? If we check for validity there, then we should never even generate a task configuration that is invalid.